packages/perseus/src/util/parse-perseus-json/general-purpose-parsers/enumeration.ts

```diff
@@ -0,0 +1,14 @@
+import type {Parser} from "../parser-types";
+
+export function enumeration<T extends readonly string[]>(
+    ...acceptedValues: T
+): Parser<T[number]> {
+    return (rawValue, ctx) => {
+        if (typeof rawValue === "string" && acceptedValues.includes(rawValue)) {
+            return ctx.success(rawValue as T[number]);
+        }
+        const expected =
+            "one of " + acceptedValues.map((v) => JSON.stringify(v)).join(", ");
+        return ctx.failure(expected, rawValue);
+    };
+}
```

Comment:
I think we might not need to check `typeof rawValue`.

Reply:
The `typeof` check is needed. `unknown` cannot be passed to either `includes` or `indexOf`.